test/query/query.test.ts

```diff
@@ -65,6 +66,29 @@ describe('query/query', () => {
       assert.equal((<SpecQueryModel>result.items[0]).specQuery.mark, 'tick');
       assert.equal((<SpecQueryModel>result.items[1]).specQuery.mark, 'point');
     });
+    it('correctly overrides properties of the provided schema when q has a schema property specified', () => {
+      const q: Query = {
+        spec: {
+          data: {
+            values: [
+              {x: '1/1/2000'}
+            ]
+          },
+          mark: 'point',
+          encodings: [
+            {channel: Channel.X, field: 'x', type: Type.TEMPORAL}
```

Comment:
change type here to enumSpec and check that the only enumerated spec is nominal.